language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -114,6 +115,17 @@ public ExtendedLanguageClient getClient() {
 
     public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
         final InitializeResult res = new InitializeResult(new ServerCapabilities());
+        res.getCapabilities().setTextDocumentSync(TextDocumentSyncKind.Full);
+
+        //Checks for instances in which the LS needs to be initiated in lightweight mode
+        if (params.getInitializationOptions() != null) {
+            JsonObject initOptions = (JsonObject) params.getInitializationOptions();
+            if (initOptions.has("enableLightWeightMode")
+                && initOptions.get("enableLightWeightMode").getAsBoolean()) {
+                return CompletableFuture.supplyAsync(() -> res);
+            }
+        }
```

Comment:
Shall we move this check to a separate method and also let's have a `enableLightWeightMode` as a constant. WDYT?